docker/image_types.yaml

```diff
@@ -17,54 +17,87 @@
   depends_on:
     - 'base-image'
 
-'coverage-builder':
-  tag: 'builders/coverage'
-  context: 'fuzzers/coverage'
-  dockerfile: 'fuzzers/coverage/builder.Dockerfile'
+'dispatcher-image':
+  tag: 'dispatcher-image'
+  context: 'docker/dispatcher-image'
+  depends_on:
+    - 'base-image'
+
+'.{fuzzer}-builder':
+  tag: 'builders/{fuzzer}'
+  context: 'fuzzers/{fuzzer}'
+  dockerfile: 'fuzzers/{fuzzer}/builder.Dockerfile'
   depends_on:
     - 'base-builder'
 
-'coverage-{benchmark}-builder':
-  tag: 'builders/coverage/{benchmark}'
+'.{fuzzer}-{benchmark}-builder':
+  tag: 'builders/{fuzzer}/{benchmark}'
   context: '.'
   dockerfile: 'docker/benchmark-builder/Dockerfile'
   build_arg:
-    - 'fuzzer=coverage'
+    - 'fuzzer={fuzzer}'
     - 'benchmark={benchmark}'
   depends_on:
-    - 'coverage-builder'
+    - '.{fuzzer}-builder'
 
-'{fuzzer}-builder':
-  tag: 'builders/{fuzzer}'
+'.{fuzzer}-{benchmark}-intermediate-runner':
+  tag: 'runners/{fuzzer}/{benchmark}-intermediate'
+  context: 'fuzzers/{fuzzer}'
+  dockerfile: 'fuzzers/{fuzzer}/runner.Dockerfile'
+  depends_on:
+    - 'base-runner'
+
+'.{fuzzer}-{benchmark}-runner':
+  tag: 'runners/{fuzzer}/{benchmark}'
+  context: '.'
+  dockerfile: 'docker/benchmark-runner/Dockerfile'
+  build_arg:
+    - 'fuzzer={fuzzer}'
+    - 'benchmark={benchmark}'
+  depends_on:
+    - '.{fuzzer}-{benchmark}-builder'
+    - '.{fuzzer}-{benchmark}-intermediate-runner'
+
+'.{benchmark}-oss-fuzz-project-builder':
+  tag: 'builders/oss-fuzz-project/{benchmark}'
+  context: 'benchmarks/{benchmark}'
+  dockerfile: 'benchmarks/{benchmark}/Dockerfile'
+
+'.{fuzzer}-{benchmark}-oss-fuzz-builder-intermediate':
+  tag: 'builders/{fuzzer}/{benchmark}-intermediate'
   context: 'fuzzers/{fuzzer}'
   dockerfile: 'fuzzers/{fuzzer}/builder.Dockerfile'
   depends_on:
-    - 'base-builder'
+    - '.{benchmark}-oss-fuzz-project-builder'
+  build_arg:
+    - 'parent_image=gcr.io/fuzzbench/builders/oss-fuzz-project/{benchmark}'
 
-'{fuzzer}-{benchmark}-builder':
+'.{fuzzer}-{benchmark}-oss-fuzz-builder':
```

Comment:
so sorry about this, I think @jonathanmetzman is almost done with getting rid of the oss-fuzz/standard difference in https://github.com/google/fuzzbench/pull/584. should we land that and remove these?

Reply:
Done